### PR TITLE
analyze: refactor version comparison code

### PR DESCRIPTION
**Pull request by Integral-Tech**

### CODE REVIEW

**Comment by poettering** [comment]
Sorry, but I don#t think this is an improvement. It makes the code longer and less readable?

File: src/analyze/analyze-compare-versions.c [hunk 1]
```diff
@@ -9,26 +9,28 @@
...
+
         /* We only output a warning on invalid version strings (instead of failing), since the comparison
          * functions try to handle invalid strings gracefully and it's still interesting to see what the
          * comparison result will be. */
-        if (!version_is_valid_versionspec(v1))
-                log_warning("Version string 1 contains disallowed characters, they will be treated as separators: %s", v1);
-        if (!version_is_valid_versionspec(v2))
-                log_warning("Version string 2 contains disallowed characters, they will be treated as separators: %s", v2);
+        for (size_t i = 0; i < sizeof(v) / sizeof(v[0]); i++) {
+                log_warning("Version string %zu contains disallowed characters, they will be treated as separators: %s",
```
Comment (poettering) [inline src/analyze/analyze-compare-versions.c]: you dropped the validation here!

**Comment by Integral-Tech** [comment]
> Sorry, but I don#t think this is an improvement. It makes the code longer and less readable?

It avoids repeating "Version string %zu contains disallowed characters, they will be treated as separators: %s".

File: src/analyze/analyze-compare-versions.c (hunk 1 above)
Comment (Integral-Tech) [reply src/analyze/analyze-compare-versions.c]: Oh, sorry :(

**Pull request closed without merging**